### PR TITLE
adding sample of stateful app

### PR DESCRIPTION
**Pull request by rsajwani**

### CODE REVIEW

File: crates/samples/echomain-stateful2/src/kvstore.rs [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+    stateful::{
+        PrimaryReplicator, StatefulServiceFactory, StatefulServicePartition, StatefulServiceReplica, Replicator,
+    },
+    stateful_types::{OpenMode, Role, Epoch, ReplicaSetConfig, ReplicaSetQuarumMode, ReplicaInfo},
+    store_types::ReplicatorSettings,
+};
+use log::info;
+use tokio::sync::oneshot::{self, Sender};
+use windows_core::{Error, HSTRING};
+mod app;
```
Comment (youyuanwu) [inline crates/samples/echomain-stateful2/src/kvstore.rs:15]: Is mod app used in this sample?

File: crates/samples/echomain-stateful/src/app.rs [hunk 2]
```diff
@@ -0,0 +1,497 @@
...
+                .as_ref()
+                .expect("not ctx")
+                .CompletedSynchronously()
+                .as_bool()
+        };
+        if !completed {
+            info!("AppInstance::EndOpen callback not completed");
+        }
+
+        //let addr = echo::get_addr(self.port_, self.hostname_.clone());
```
Comment (youyuanwu) [inline crates/samples/echomain-stateful/src/app.rs:374]: Clean up unused comments.

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,80 +1,94 @@
-cmake_minimum_required(VERSION 3.14)
```
Comment (youyuanwu) [inline CMakeLists.txt:1]: Fix line endings or indent?

**Comment by youyuanwu** [comment]
Describe stateful service 1 and stateful service 2 and what their differences are.

**Review by youyuanwu — CHANGES_REQUESTED** [review]
Please fix some minor comments.

File: crates/samples/echomain-stateful/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+[dependencies.fabric_base]
+path = "../../fabric/base"
+features = [
+    "implement",
+    "ServiceFabric",
+    "ServiceFabric_FabricCommon",
+    "ServiceFabric_FabricCommon_FabricRuntime"
+]
+
+[dependencies.fabric_rs]
```
Comment (youyuanwu) [inline crates/samples/echomain-stateful/Cargo.toml:28]: If you want to demo COM usage you should ideally remove this dependency

File: crates/samples/echomain-stateful2/manifests/ServiceManifest.Linux.xml [hunk 5]
```diff
@@ -0,0 +1,33 @@
+<!--Sample XML file generated by XMLSpy v2011 rel. 2 sp1 (http://www.altova.com)-->
```
Comment (youyuanwu) [inline crates/samples/echomain-stateful2/manifests/ServiceManifest.Linux.xml:1]: No longer needed.

File: crates/samples/echomain-stateful/Cargo.toml (hunk 4 above)
Comment (rsajwani) [reply crates/samples/echomain-stateful/Cargo.toml:28]: I can see even echomain is using use fabric_rs::{AsyncContext, StringResult}; ... seems like there are some common namespaces

**Review by youyuanwu — APPROVED** [review]

**Pull request merged**